### PR TITLE
Version settings for Python 3.12

### PR DESCRIPTION
**Pull request by loechel**

### CODE REVIEW

**Comment by loechel** [comment]
so the changes in Python 3.12 especially on the Class and Function Definition has to been looked in before makeing an official Release that supports 3.12

File: setup.py [hunk 1]
```diff
@@ -52,6 +52,7 @@ def read(*rnames):
         'Programming Language :: Python :: 3.9',
         'Programming Language :: Python :: 3.10',
         'Programming Language :: Python :: 3.11',
+        'Programming Language :: Python :: 3.12',
```
Comment (icemac) [inline setup.py:55]: This is too early, until we did a security review of the Python 3.12 features, we cannot claim to support Python 3.12.

**Review by icemac — CHANGES_REQUESTED** [review]
I mostly the proposed changes.

File: setup.py (hunk 1 above)
Comment (davisagli) [reply setup.py:55]: @icemac @d-maurer did an audit of the Python 3.12 features here: https://github.com/zopefoundation/RestrictedPython/issues/246#issuecomment-1774594376

Can we go ahead?

**Review by dataflake — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (icemac) [reply setup.py:55]: yes.

**Review by icemac — APPROVED** [review]

**Pull request merged**